### PR TITLE
Fix permissions handling for annotation deletion events

### PR DESCRIPTION
**Pull request by robertknight**
Previously annotation events were only delivered to a WebSocket client
if the client had read permissions for the annotation. For annotation
deletion events however, nobody has annotation read permissions
as determined by `AnnotationContext.__acl__`. As a result, delete
notifications were never delivered.

Add custom logic for the delete event which delivers the notifications
of deletions to anyone who can read the group containing the annotation,
if the annotation is shared, or the user who created the annotation, if
it is private.

This commit also removes wasted overhead in serializing the annotation
to JSON if the annotation has been deleted, since the payload in the
notification only contains the annotation ID.

### CODE REVIEW

**Comment by jon-betts** [comment]
When I've been trying this manually adding an annotation for myself and deleting it, I've been seeing these deletes work fine. Which means it's a bit tricky for me to say if this is working or not.

Would you be expecting to see any deletes working? If there are some deletes that should, and some that should could you add some details about how to test the difference?

File: tests/h/streamer/messages_test.py [hunk 1]
```diff
@@ -363,6 +366,55 @@ def test_sends_if_in_group(
...
+        settings = {"foo": "bar"}
+
+        def principals(resource, permission):
+            if (
+                resource == annotation_resource.return_value.group
+                and permission == "read"
+            ):
+                return ["group:foobar"]
+            else:
+                return []
```
Comment (jon-betts) [inline tests/h/streamer/messages_test.py:407]: There's a bit too much conditionality here to be clear about what's going on. I was expecting to see a `principals_allowed_by_permission.assert_called_once_with(...)` line here.

If that was combined with a canned response, I don't think we'd need this to be so active.

**Review by jon-betts — CHANGES_REQUESTED** [review]
I've put some suggestions below, but my overall feeling is that we are replicating something like the `AnnotationContext.__acl__` method in-line here. This suggests to me that perhaps we don't want an `AnnotationContext` at all / we aren't in that context.

Would this be more consistent if we had an `AnnotationNotificationContext` with an '__acl__` method that contained this info? 

`AnnotationContext` is quite right when it says nobody has permission to look at a deleted annotation etc. but our new context could rightfully suggest you have the right to be notified about it.

**Comment by robertknight** [comment]
This was fixed a different way via https://github.com/hypothesis/h/pull/6294.

**Pull request closed without merging**